File: fedorova-t.v-kt-41-22/Database/Configurations/LoadConfiguration.vb
Imports fedorova_t.v_kt_41_22.Models
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders


Namespace fedorova_t.v_kt_41_22.Database.Configurations
    Public Class LoadConfiguration
        Implements IEntityTypeConfiguration(Of Load)
        Public Sub Configure(builder As EntityTypeBuilder(Of Load)) Implements IEntityTypeConfiguration(Of Load).Configure
            builder.ToTable("Loads")
            builder.HasKey(Function(l) l.Id)
            builder.Property(Function(l) l.Hours).IsRequired()

            builder.HasOne(Function(l) l.Teacher).WithMany(Function(t) t.Loads).HasForeignKey(CType(Function(l) l.TeacherId, Expressions.Expression(Of Func(Of Load, Object?)))).OnDelete(DeleteBehavior.Cascade)

            builder.HasOne(Function(l) l.Discipline).WithMany(Function(d) d.Loads).HasForeignKey(CType(Function(l) l.DisciplineId, Expressions.Expression(Of Func(Of Load, Object?)))).OnDelete(DeleteBehavior.Restrict)
        End Sub
    End Class
End Namespace

File: fedorova-t.v-kt-41-22/Database/Configurations/LoadConfiguration.cs
using fedorova_t.v_kt_41_22.Models;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using fedorova_t.v_kt_41_22.Database.Helpers;

namespace fedorova_t.v_kt_41_22.Database.Configurations
{
    public class LoadConfiguration : IEntityTypeConfiguration<Load>
    {
        private const string TableName = "Loads";
        public void Configure(EntityTypeBuilder<Load> builder)
        {
            builder.HasKey(p => p.Id)
               .HasName($"pl_{TableName}_load_id");

            //Автоинкрементация
            builder.Property(p => p.Id)
                .ValueGeneratedOnAdd()
                .HasColumnName("Load_Id")
                .HasComment("Id нагрузки");

            // Настройка связи c teachers
            builder.Property(p => p.TeacherId)
                .IsRequired()
                .HasColumnName("Teacher_Id")
                .HasComment("Id преподавателя")
                .HasColumnType(ColumnType.Int);


            builder.HasOne(p => p.Teacher)
                .WithMany()
                .HasForeignKey(p => p.TeacherId)
                .HasConstraintName("fk_f_teacher_id")
                .OnDelete(DeleteBehavior.Cascade);


            builder.ToTable(TableName)
                .HasIndex(p => p.TeacherId, $"idx_{TableName}_fk_f_teacher_id");

            builder.Navigation(p => p.Teacher)
             .AutoInclude();

            // Настройка связи c дисциплинами
            builder.Property(p => p.DisciplineId)
                .IsRequired()
                .HasColumnName("Discipline_Id")
                .HasComment("Id дисциплины")
                .HasColumnType(ColumnType.Int);


            builder.HasOne(p => p.Discipline)
                .WithMany()
                .HasForeignKey(p => p.DisciplineId)
                .HasConstraintName("fk_f_discipline_id")
                .OnDelete(DeleteBehavior.Cascade); //Пересмотреть

            builder.ToTable(TableName)
                .HasIndex(p => p.DisciplineId, $"idx_{TableName}_fk_f_discipline_id");

            builder.Navigation(p => p.Discipline)
                .AutoInclude();

            builder.Property(p => p.Hours)
                .IsRequired()
                .HasColumnName("Hours")
                .HasComment("Количество часов нагрузки")
                .HasColumnType(ColumnType.Int);

            builder.ToTable(TableName);
        }
    }
    }
